pkg/mimirpb/custom.go

```diff
@@ -44,6 +47,132 @@ func (h Histogram) IsGauge() bool {
 	return h.ResetHint == Histogram_GAUGE
 }
 
+// ReduceResolution will reduce the resolution of the histogram by one level.
+// Returns the resulting bucket count and an error if the histogram is not
+// possible to reduce further.
+func (h *Histogram) ReduceResolution() (int, error) {
+	if h.IsFloatHistogram() {
+		return h.reduceFloatResolution()
+	}
+
+	return h.reduceIntResolution()
+}
+
+func (h *Histogram) reduceFloatResolution() (int, error) {
+	if h.Schema == -4 {
```

Comment:
Can we have a meaningful constant for this somewhere?

Reply:
Added definition in the same file. I looked again, but it's never exported from Prometheus, so I have to make a PR in Prometheus first to solve it...